app/src/main/kotlin/com/hippo/ehviewer/ui/screen/HistoryScreen.kt

```diff
@@ -52,53 +58,69 @@ import com.hippo.ehviewer.util.FavouriteStatusRouter
 import com.ramcosta.composedestinations.annotation.Destination
 import com.ramcosta.composedestinations.navigation.DestinationsNavigator
 import eu.kanade.tachiyomi.util.lang.launchIO
+import kotlin.math.roundToInt
 import kotlinx.coroutines.delay
-import kotlinx.coroutines.launch
 
 @Destination
 @Composable
 fun HistoryScreen(navigator: DestinationsNavigator) = composing(navigator) {
-    val scrollBehavior = TopAppBarDefaults.pinnedScrollBehavior()
+    val title = stringResource(id = R.string.history)
+    val hint = stringResource(R.string.search_bar_hint, title)
+
+    var searchBarOffsetY by remember { mutableIntStateOf(0) }
+    var keyword by rememberSaveable { mutableStateOf("") }
+
+    val density = LocalDensity.current
     val historyData = rememberInVM {
         Pager(config = PagingConfig(pageSize = 20, jumpThreshold = 40)) {
-            EhDB.historyLazyList
+            if (keyword.isNotEmpty()) {
+                EhDB.searchHistory(keyword)
+            } else {
+                EhDB.historyLazyList
+            }
         }.flow.cachedIn(viewModelScope)
     }.collectAsLazyPagingItems()
     FavouriteStatusRouter.Observe(historyData)
-    Scaffold(
-        modifier = Modifier.nestedScroll(scrollBehavior.nestedScrollConnection),
-        topBar = {
-            TopAppBar(
-                title = { Text(text = stringResource(id = R.string.history)) },
-                navigationIcon = {
-                    val drawerState = LocalNavDrawerState.current
-                    IconButton(onClick = { launch { drawerState.open() } }) {
-                        Icon(imageVector = Icons.Default.Menu, contentDescription = null)
-                    }
-                },
-                actions = {
-                    IconButton(onClick = {
-                        launchIO {
-                            awaitPermissionOrCancel(
-                                confirmText = R.string.clear_all,
-                                text = { Text(text = stringResource(id = R.string.clear_all_history)) },
-                            )
-                            EhDB.clearHistoryInfo()
-                        }
-                    }) {
-                        Icon(imageVector = Icons.Default.ClearAll, contentDescription = null)
-                    }
-                },
-                scrollBehavior = scrollBehavior,
-            )
+    SearchBarScreen(
+        title = title,
+        searchFieldHint = hint,
+        onApplySearch = {
+            keyword = it
+            historyData.refresh()
+        },
+        onSearchExpanded = {},
+        onSearchHidden = {},
+        searchBarOffsetY = { searchBarOffsetY },
+        trailingIcon = {
+            IconButton(onClick = {
+                launchIO {
```

Comment:
Maybe we don't have to use `Dispatchers.IO`, as DB operations are suspend and executed in certain Executors now